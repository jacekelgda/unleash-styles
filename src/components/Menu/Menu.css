.Menu {
	position: relative;
	width: 54px;
	user-select: none;
}
.MenuList {
	list-style: none;
	display: none;
	position: absolute;
	z-index: 10;
	top: 53px;
	right: 0;
	margin: 0;
	padding: 30px;
	background-color: #fff;
	box-shadow: 0 10px 30px rgba(220,220,220,.5);
}

.MenuList_active {
	display: block;
}
