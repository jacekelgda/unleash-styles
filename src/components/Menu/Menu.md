Menu:

    <Menu menuButton={<Avatar type="small" label="NO" />}>
      <MenuItem label="nikita" type="important" />
      <MenuItem label="nikita.ormonov@eleken.co" type="email" />
      <MenuItem label="edit profile" type="link" />
      <MenuItem label="sign out" type="link" />
    </Menu>
