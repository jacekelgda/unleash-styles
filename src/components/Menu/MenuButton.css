.MenuButton {
  position: absolute;
  content: url(../../assets/three-dots.svg);
  border-radius: 50%;
  width: 20px;
  height: 20px;
  top: 16px;
  right: 18px;
  cursor: pointer;
  display: block;
}
