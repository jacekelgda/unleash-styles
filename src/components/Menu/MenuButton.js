import React from 'react'
import styles from './MenuButton.css'

const MenuButton = () => (
  <div className={styles.MenuButton} />
)

export default MenuButton
