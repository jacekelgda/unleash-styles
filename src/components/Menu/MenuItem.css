.MenuItem {
	line-height: 34px !important;
  font: 16px/1.25 'Open Sans', sans-serif;
}

.MenuItemName {
  font: 700 18px 'Montserrat', sans-serif;
	text-transform: uppercase;
	color: #242424;
}

.MenuItemEmail {
	font-size: 16px;
	color: #b8b7bc;
	margin-bottom: 25px;
	padding-bottom: 25px;
	border-bottom: 1px solid #e8e8e8;
}
.MenuItemLink {
	font: 600 14px 'Montserrat', sans-serif;
	text-decoration: none;
	text-transform: uppercase;
	color: #b8b7bc;
}
