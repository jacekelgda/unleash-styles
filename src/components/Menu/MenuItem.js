import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './MenuItem.css'

const TYPE_LINK = 'link'
const TYPE_IMPORTANT = 'important'
const TYPE_EMAIL = 'email'

const typeStyle = {
  link: styles.MenuItemLink,
  important: styles.MenuItemName,
  email: styles.MenuItemEmail
}

class MenuItem extends Component {
  renderLink(label) {
    return(
      <a href="#" className={styles.MenuItemLink}>{label}</a>
    )
  }

  render() {
    const { label, type } = this.props
    return (
      <li className={`${styles.MenuItem} ${typeStyle[type]}`}>
        {type == TYPE_LINK ? this.renderLink(label) : label}
      </li>
    )
  }
}

MenuItem.propTypes = {
  /** Text on menu item */
  label: PropTypes.string,
  /** Type of menu item: link, important, email */
  type: PropTypes.string,
}

export default MenuItem
