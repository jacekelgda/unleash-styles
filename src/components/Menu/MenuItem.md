Menu item:

    <Menu alwaysOpened="true">
      <MenuItem label="Item" type="important"/>
    </Menu>
