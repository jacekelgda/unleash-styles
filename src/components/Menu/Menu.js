import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import styles from './Menu.css'
import MenuButton from './MenuButton'

/**
 * Menu component can be used to build list menus. They are hidden by defaul
 * but with menu button click they show up
 */
class Menu extends Component {
  static defaultProps: Object

  state: {
    isShowing: boolean
  }

  constructor (props: Object) {
    super(props)

    this.state = {
      isShowing: false
    }
  }

  handleClick () {
    this.setState({
      isShowing: !this.state.isShowing
    })
  }

  render () {
    const { menuButton, isAlwaysOpened } = this.props
    const listClasses = classNames(
      styles.MenuList,
      (this.state.isShowing || isAlwaysOpened) && styles.MenuList_active
    )

    return (
      <nav className={styles.Menu}>
        <div onClick={() => this.handleClick()}>{menuButton}</div>
        <ul className={listClasses}>
          {this.props.children}
        </ul>
      </nav>
    )
  }
}

Menu.defaultProps = {
  menuButton: (<MenuButton />),
  isAlwaysOpened: false
}

Menu.propTypes = {
  /** Component that triggers opening and closing menu */
  menuButton: PropTypes.node,
  /** Set true if you would like menu to be opened always */
  isAlwaysOpened: PropTypes.bool
}

export default Menu
