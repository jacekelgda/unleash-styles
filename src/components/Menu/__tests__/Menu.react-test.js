import React from 'react'
import renderer from 'react-test-renderer'
import Menu from '../Menu'
import MenuItem from '../MenuItem'
import Avatar from '../../Avatar/Avatar'

test('Menu renders correctly', () => {
  const tree = renderer.create(
    <Menu/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Menu renders with MenuItems', () => {
  const tree = renderer.create(
    <Menu>
      <MenuItem label="test" type="important" />
      <MenuItem label="test2" type="emial" />
    </Menu>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Menu renders with MenuItems and custom menu button', () => {
  const tree = renderer.create(
    <Menu menuButton={<Avatar label="test" />}>
      <MenuItem label="test@test.com" type="email" />
      <MenuItem label="test2" type="important" />
    </Menu>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
