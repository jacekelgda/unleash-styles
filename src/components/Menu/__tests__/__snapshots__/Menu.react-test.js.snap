// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<nav
  className="Menu"
>
  <div
    onClick={[Function]}
  >
    <div
      className="MenuButton"
    />
  </div>
  <ul
    className="MenuList false"
  />
</nav>
`;

exports[`Menu renders with MenuItems 1`] = `
<nav
  className="Menu"
>
  <div
    onClick={[Function]}
  >
    <div
      className="MenuButton"
    />
  </div>
  <ul
    className="MenuList false"
  >
    <li
      className="MenuItem MenuItemName"
    >
      test
    </li>
    <li
      className="MenuItem undefined"
    >
      test2
    </li>
  </ul>
</nav>
`;

exports[`Menu renders with MenuItems and custom menu button 1`] = `
<nav
  className="Menu"
>
  <div
    onClick={[Function]}
  >
    <div>
      <input
        className="AvatarButton"
        id="avatar_button"
        onClick={undefined}
        type="checkbox"
      />
      <label
        className="AvatarButtonLabel"
        for="avatar_button"
      >
        test
      </label>
    </div>
  </div>
  <ul
    className="MenuList false"
  >
    <li
      className="MenuItem MenuItemEmail"
    >
      test@test.com
    </li>
    <li
      className="MenuItem MenuItemName"
    >
      test2
    </li>
  </ul>
</nav>
`;
